docs: sync Verawood dev/operator release notes with wiki updates (#1486)
openedx/docs.openedx.org: source/community/release_notes/verawood/dev_op_release_notes.rst

Reviewer: ```suggestion
See the `Build Test Release project board <https://github.com/orgs/openedx/projects/28/views/19>`_ for known open issues.
```

  implementation. In either case, the third-party implementation must be
  removed from the ``xblock.v1`` entrypoint, otherwise you will see an
  ``AmbiguousPluginError``. Additionally, you will want to enable the
  ``legacy_studio.pdf_editor`` toggle to allow the third-party pdf
  implementation to render its editor.

* **New Instructor Dashboard MFE is the default experience.** The new
  instructor dashboard MFE will be the default experience in Verawood. The
  ``instructor.legacy_instructor_dashboard`` waffle flag will disable the
  redirection that makes the new frontend app the default experience and
  revert to the legacy instructor dashboard.

  If you were using the
  ``org.openedx.learning.instructor.dashboard.render.started.v1`` filter to
  modify the instructor dashboard tabs, you can achieve a similar outcome by
  using ``org.openedx.learning.instructor.dashboard.tabs.generated.v1``, which
  will give you access to the API endpoint currently consumed to render the
  tabs on the frontend app. You'll also need to add a widget to the proper
  slot to render the content. See the instructor dashboard MFE README for
  more details.

> 

Reviewer: Can we include this commit? Thanks! https://github.com/mariajgrimaldi/docs.openedx.org/commit/2265e656e147bb60ee60f013867c87f70c601ddd

(we missed adding [this](https://openedx.atlassian.net/wiki/spaces/OEPM/pages/6331662350/RBAC+AuthZ+for+Course+Authoring+-+Operator+Release+Notes) to the confluence page after leaving this comment https://openedx.atlassian.net/wiki/spaces/COMM/pages/5331222534/Verawood+-+Operator+Release+Notes?focusedCommentId=6544097307 - so we're adding directly it)

Author: Thank you!

==========================

If you deploy Open edX using something other than Tutor, start by looking at
`frontend-template-site <https://github.com/openedx/frontend-template-site>`_.
This is the canonical representation of how a frontend-base site should be
configured. Operators are meant to fork and modify it as required for their
own instances. Everything in it is considered user-configuration, including
``index.html``.

Loading External Scripts
*************************

It is now possible to configure either MFEs or frontend apps to load
arbitrary external scripts, such as ones used to configure hosted analytics
services. Where before only Google Analytics was supported, now operators
are free to load whatever external service they need. See `the
corresponding tutor-mfe documentation
<https://github.com/overhangio/tutor-mfe/tree/main#configuring-external-scripts>`_.

For an example implementation, refer to a sample Tutor plugin that loads
Google Analytics dynamically on both MFEs and frontend apps:
`tutor-contrib-google-analytics
<https://github.com/openedx/openedx-tutor-plugins/tree/main/plugins/tutor-contrib-google-analytics>`_.

Settings and Toggles

> ********************

Reviewer: Can we include this section of migrations to be aware of? https://github.com/mariajgrimaldi/docs.openedx.org/commit/43bf8b99670a281315b3b2f6f9a5487e1864921c

FYI @bradenmacdonald

Reviewer 2: Looks like this was added above in section "Migrations to be aware of" ?

  redirection that makes the new frontend app the default experience and
  revert to the legacy instructor dashboard.

  If you were using the
  ``org.openedx.learning.instructor.dashboard.render.started.v1`` filter to
  modify the instructor dashboard tabs, you can achieve a similar outcome by
  using ``org.openedx.learning.instructor.dashboard.tabs.generated.v1``, which
  will give you access to the API endpoint currently consumed to render the
  tabs on the frontend app. You'll also need to add a widget to the proper
  slot to render the content. See the instructor dashboard MFE README for
  more details.

* **RBAC AuthZ for Course Authoring (opt-in).** Verawood introduces a new,
  opt-in, RBAC-based authorization system for course authoring in Studio,
  powered by `openedx-authz <https://github.com/openedx/openedx-authz>`_.
  This replaces the legacy ``CourseAccessRole``-based permission model with a
  more granular, role-based system. See `ADR: AuthZ for Course Authoring
  Implementation Plan
  <https://github.com/openedx/openedx-authz/blob/main/docs/decisions/0009-authz-for-course-authoring.rst>`_.

  This flag will be turned on by default in the next release, Willow, and

>   the legacy permission system will be removed in Xylon. To set up an

Reviewer 2: ```suggestion
  the legacy permission system will be removed in Xylon (June 2027). To set up an
```

information on what motivated this change, see `OEP-65
<https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0065-arch-frontend-composability.html>`_.

In practice, frontend apps are NPM packages that plug into a single
frontend-base site. Verawood ships with four of these:

* ``@openedx/frontend-app-authn``: replaces the Authn MFE; *disabled by
  default*
* ``@openedx/frontend-app-learner-dashboard``: replaces the Learner
  Dashboard MFE; *disabled by default*
* ``@openedx/frontend-app-instructor-dashboard``: replaces the legacy
  instructor dashboard; *enabled by default*
* ``@openedx/frontend-app-notifications``: ports
  ``@edx/frontend-plugin-notifications`` to the frontend-base ecosystem;
  *enabled by default*

The tutor-mfe implementation allows each of these to be easily enabled or
disabled without requiring a rebuild of the Tutor image, as documented in
the `tutor-mfe README
<https://github.com/overhangio/tutor-mfe/tree/main#frontend-base-site>`_.
When an app that replaces an existing MFE is enabled (such as Authn or
Learner Dashboard), its MFE counterpart is automatically disabled.

> It is expected that by Xylon (the Open edX release following Willow), all

Reviewer 2: ```suggestion
It is expected that by Xylon (June 2027), all
```

  <https://github.com/openedx/frontend-base/blob/main/docs/how_tos/theming.md>`_
* `docs.openedx.org PR #1459
  <https://github.com/openedx/docs.openedx.org/pull/1459>`_
* `Migrate a frontend app how-to
  <https://github.com/brian-smith-tcril/frontend-base/blob/main/docs/how_tos/migrate-frontend-app.md>`_

As a migration aid for unconverted legacy plugins, a `frontend-base-compat
<https://github.com/openedx/frontend-base-compat>`_ shim lets legacy
``env.config.jsx`` configuration run on top of frontend-base sites. It is
also implemented by tutor-mfe: refer to `the corresponding section
<https://github.com/overhangio/tutor-mfe/tree/main#compat-shim-for-fpf-built-plugins>`_
of its documentation for information on how to use it.

Finally, note that tutor-mfe's frontend-base implementation automatically
makes use of the new ``/api/frontend_site_config/v1/`` runtime configuration
endpoint, which converts existing ``MFE_CONFIG`` and ``MFE_CONFIG_OVERRIDES``
to frontend-base's ``SiteConfig`` structure. You can find out more `here
<https://github.com/overhangio/tutor-mfe/tree/main#backward-compatibility-with-mfe_config>`_.

What if I don't use Tutor?
==========================

> If you deploy Open edX using something other than Tutor, start by looking at

Reviewer 2: ```suggestion
If you deploy your Open edX instance using something other than Tutor, start by looking at
```

@feoh could you please audit these release notes to ensure you're not using "Open edX" as a noun in other places?
